### PR TITLE
feat(deposit-track): expose endpoint

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR adds an endpoint to track deposits given a `originChainId` and a `depositHash`. Because a deposit hash can include multiple deposit records, an array is returned.

### CODE REVIEW

File: api/track-deposit.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  handleErrorCondition,
+  positiveIntStr,
+  resolveFillStatusOfDeposits,
+  sendResponse,
+  validTxnHash,
+} from "./_utils";
+
+const TrackDepositQueryParamsSchema = object({
+  originChainId: positiveIntStr(),
+  depositHash: validTxnHash(),
```
Comment (nicholaspai) [inline api/track-deposit.ts:16]: what if user wants to just specify `depositId`?

Comment (nicholaspai) [reply api/track-deposit.ts:16]: i don’t think specifying the transaction has is the right approach, it doesnt’ handle the case where there are multiple deposits in the same hash. Moreover, searching for a log seems inefficient.
Instead, why not make caller pass in the keccak256(deposit) i.e. the keccak256DepositHash

**Comment by nicholaspai** [comment]
let's close this in favor of using the scraper API

**Comment by james-a-morris** [comment]
Closing as this was absorbed by the scraper work

**Pull request closed without merging**